gpu-alloc/src/block.rs

```diff
@@ -158,37 +159,33 @@ impl<M> MemoryBlock<M> {
             "`offset + size` is out of memory block bounds"
         );
 
-        let ptr = match self.flavor {
-            MemoryBlockFlavor::Dedicated => {
+        let ptr = match &mut self.flavor {
```

Comment:
fwiw, in gfx/wgpu code we prefer to not match on references. It's your call here, so no objections, just fyi

Reply:
And use `ref` and `ref mut` in patterns? As I understand it, the semantics is the same. Just choice of style and MSRV

Reply:
Right. My preference is to make this explicit at the match pattern side, i.e. knowing the type being matched it's easier to see what is referenced, how deep, what's moved, or copied otherwise, based on those `ref`.